doc/decisions/explanations.md

```diff
@@ -0,0 +1,91 @@
+# Explanations
+
+## Problem
+
+Clearly define:
+
+- the context in which the problem exists
+- your observation of the problem
+- ideally an example for that problem
+
+## Constraints
+
+List all constraints given by:
+
+- use cases
+- requirements
+- [Elektra's goals](/doc/GOALS.md)
+- guidelines, e.g. [documentation guidelines](/doc/contrib/documentation.md)
+- standards, e.g. [C99](https://www.open-std.org/jtc1/sc22/wg14/)
+- other decisions
+- the scope (i.e. describe what shouldn't change)
+
+> Note:
+> The decision (but not necessarily the considered alternatives) must fulfill all constraints.
+
+## Assumptions
+
+Often we are quite blind and do not see assumptions we have.
+So this section needs special care and honesty.
+Assumptions is what we believe to be true but do not or cannot really know, e.g.:
+
+- what users will accept
+- perceived usability
+- if the implementation will be faster/slower
+- estimations of costs
+- problems/risks that might turn up
+
+> Note:
+> The decision (but not necessarily the considered alternatives) must fulfill all assumptions.
```

Comment:
```suggestion
> The decision (but not necessarily the considered alternatives) must not break any assumptions.
```

There could be assumptions that are needed to make a considered alternative work, but are not needed for the final decision.

For example: "We assume algorithm X is faster than algorithm Y which we currently use". If the final decision stays with Y, but an alternative uses X, that should be fine, because it doesn't make things worse.